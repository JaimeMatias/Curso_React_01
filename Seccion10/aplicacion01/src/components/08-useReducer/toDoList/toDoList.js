import React from 'react'
import { ToDoListItem } from '../toDoListItem/toDoListItem'

export const ToDoList = ({ input, handleDelete, handleToggle }) => {
    // si uso {} todo lo que defina en el centro, va a ser considerado como elementos de un objeto
    //Si yo defino valores afuera de los corchetes, van a ser considerado como valores de entrada (no pueden ser funciones)
    return (
        <div className='col-7'>
            TODOS
            {/*TodoListItem,Todo,index,handleDelete,handleToggle*/}
            <ul className='list-group list-group.flush'>
                {
                    input.map((element, i) => {
                        //TodoListItem, todo,index, handleDelete,handelToggle
                        return (

                            <ToDoListItem key={element.id}element={element} contador={i} handleDelete={handleDelete} handleToggle={handleToggle} />



                        )
                    })
                }
            </ul>
        </div>
    )
}

