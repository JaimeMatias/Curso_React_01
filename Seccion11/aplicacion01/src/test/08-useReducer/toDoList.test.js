import React from 'react';
import { shallow } from "enzyme"
import { demoTodos } from '../fixtures/demoTodos';
import { ToDoList } from '../../components/08-useReducer/toDoList/toDoList';

describe('Pruebas en <TodoList />>', () => {
    const handleToggle=jest.fn();
    const handleDelete=jest.fn();

    const wrapper = shallow(<ToDoList 
        input ={demoTodos}
        handleDelete={handleDelete}
         handleToggle={handleToggle}
      />)

test('debe de mostrarse corectamente ', () => {

    expect(wrapper).toMatchSnapshot()

})

test('debe de tener dos  <TodoListItem> ', () => {

expect(wrapper.find('ToDoListItem').length).toBe(demoTodos.length);
expect(wrapper.find('ToDoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function))
})


})
