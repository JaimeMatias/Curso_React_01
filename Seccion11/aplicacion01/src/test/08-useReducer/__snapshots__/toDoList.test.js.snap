// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList />> debe de mostrarse corectamente  1`] = `
<div
  className="col-7"
>
  TODOS
  <ul
    className="list-group list-group.flush"
  >
    <ToDoListItem
      contador={0}
      element={
        Object {
          "desc": "Aprender React",
          "done": false,
          "id": 1,
        }
      }
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      key="1"
    />
    <ToDoListItem
      contador={1}
      element={
        Object {
          "desc": "Aprender Mongo",
          "done": false,
          "id": 2,
        }
      }
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      key="2"
    />
  </ul>
</div>
`;
